optimum/intel/ipex/inference.py

```diff
@@ -99,18 +189,19 @@ def __enter__(self):
                             with torch.cpu.amp.autocast(enabled=(self._dtype == torch.bfloat16)), torch.no_grad():
                                 if self._model.tokenizer is not None and self._jit:
                                     try:
-                                        jit_inputs = []
-                                        dummy_input = self._model.tokenizer("")
-                                        for key in dummy_input:
-                                            jit_inputs.append(torch.ones((1, len(dummy_input[key])), dtype=torch.long))
+                                        jit_inputs = prepare_jit_inputs(self._model.model, self._model.task)
+                                        torch._C._jit_set_profiling_mode(False)
                                         model = torch.jit.trace(model, jit_inputs, strict=False)
                                         model = torch.jit.freeze(model)
                                         model(*jit_inputs)
                                         model(*jit_inputs)
                                     except Exception as e:
                                         logger.warning(f"failed to use PyTorch jit mode due to: {e}.")
                                 # Patching model with the new one
-                                self._model.model = _ModelFallbackWrapper(model, self._original)
+                                if "generation" in self._model.task:

```

Comment:
If "text2text-generation" task is also concerned, we should add some tests as well, what do you think ?

Reply:
I have changed it. We only consider text-generation for now, but the text2text-generation is on the way.

Reply:
Thanks !